vnext/Microsoft.ReactNative.IntegrationTests/ReactNativeHostTests.cpp

```diff
@@ -81,13 +81,33 @@ TEST_CLASS (ReactNativeHostTests) {
 
     auto reactNativeHost = TestReactNativeHostHolder(L"ReactNativeHostTests", [](ReactNativeHost const &host) noexcept {
       host.PackageProviders().Append(winrt::make<TestPackageProvider>());
+      return true;
     });
 
     TestEventService::ObserveEvents(
-        {TestEvent{"initialize", nullptr},
-         TestEvent{"start tests", nullptr},
+        {TestEvent{"TestHostModule::Initialize", nullptr},
+         TestEvent{"TestHostModule::startTests", nullptr},
          TestEvent{"call addValues", JSValueArray{4, 7}},
-         TestEvent{"return result", 11}});
+         TestEvent{"TestHostModule::ReturnResult", 11}});
+  }
+
+  TEST_METHOD(HostReloadedAsyncActionCompletedEvent) {
+    TestEventService::Initialize();
+
+    auto reactNativeHost = TestReactNativeHostHolder(L"ReactNativeHostTests", [](ReactNativeHost const &host) noexcept {
+      host.ReloadInstance().Completed([](auto const &, winrt::Windows::Foundation::AsyncStatus status) mutable {
+        if (status == winrt::Windows::Foundation::AsyncStatus::Completed) {
+          TestEventService::LogEvent("InstanceLoaded::Completed", nullptr);
+        } else if (status == winrt::Windows::Foundation::AsyncStatus::Canceled) {
+          TestEventService::LogEvent("InstanceLoaded::Canceled", nullptr);
+        } else {
+          TestEventService::LogEvent("InstanceLoaded::Failed", nullptr);
+        }
+      });
+      return false; // Already called ReloadInstance, so we dont need the holder to load the instance
```

Comment:
> return false [](http://example.com/codeflow?start=5&length=13)

I do not like the idea that now we have to return a boolean. I could not understand the meaning of the returned value some time. It would be better to add an optional parameter to control this behaviour instead.
The point is that the TestReactNativeHostHolder does the initializaiton of the RNH. Before it was a fixed set of steps. This time we want to control one aspect of it. In future there can be other aspects. Thus, having an overload that receives a struct with options before the lambda arg would be a better option.
I believe that the returned value from the lambda is not that expressive and makes code that does not need to control that behaviour unnecessary more complex.

Reply:
Agreed.  I'll change it in main. -- I'll keep the backports as is.

Reply:
Thank you! Sorry for not reviewing your code yesterday...